common/flagdinjector/flagdinjector.go

```diff
@@ -29,8 +29,7 @@ const (
 type IFlagdContainerInjector interface {
 	InjectFlagd(
 		ctx context.Context,
-		objectMeta *metav1.ObjectMeta,
-		podSpec *corev1.PodSpec,
+		pod *corev1.Pod,
```

Comment:
is there a chance we can keep using `PodSpec` here? Otherwise the `Flagd` CRD controller which is also using this interface for inserting a flagd container into the PodSpec of a `Deployment` will not be able to use this easily anymore (at least it will be a bit awkward to use in that case)

Reply:
hmm yes I see, I guess I will revert the change